### DISCUSSION TITLE
[feg] Connect FGW to HSS

### DISCUSSION THREAD

**Discussion by MGAbdElKhalek** [magma/magma#5846] [Q&A] [answered]
I'm trying to connect the Federation GW to the OAI hss_sim and when it fails. Both elements are in the same Virtual machine.
`/var/opt/magma/docker$ docker logs s6a_proxy`

> I0331 16:20:15.706273       1 service_config.go:169] Successfully loaded '::service_registry' service configs from '/etc/magma/service_registry.yml'
> I0331 16:20:15.706714       1 service_config.go:169] Successfully loaded '::service_registry' service configs from '/etc/magma/service_registry.yml'
> W0331 16:20:16.457636       1 service_config.go:197] Cannot find '/etc/magma/configs/feg/s6a_proxy.yml': stat /etc/magma/configs/feg/s6a_proxy.yml: no such file or directory, or Legacy Service Registry Configuration: '/etc/magma/feg/s6a_proxy.yml': stat /etc/magma/feg/s6a_proxy.yml: no such file or directory
> E0331 16:20:16.468228       1 service_config.go:167] Error Loading feg::s6a_proxy configs from '/etc/magma/feg/s6a_proxy.yml': open /etc/magma/feg/s6a_proxy.yml: no such file or directory
> W0331 16:20:16.468438       1 service.go:103] Failed to load config for service S6A_PROXY: open /etc/magma/feg/s6a_proxy.yml: no such file or directory
> I0331 16:20:16.468807       1 service_config.go:111] Successfully loaded structured '::magmad' service configs from '/etc/magma/magmad.yml'
> 2021/03/31 16:20:16 Loaded s6a_proxy configs: {LogLevel:INFO Server:protocol:"tcp" address:" 127.0.0.34:3868" local_address:" 127.0.0.36:3869" product_name:"HssDiameter" realm:"HssOriginRealm.com" host:"HssOriginHost.com" dest_realm:"magma.com" dest_host:"magma-fedgw.magma.com"  RequestFailureThreshold:0 MinimumRequestThreshold:0 PlmnIds:[] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}
> 2021/03/31 16:20:16 Using default value: HssOriginHost.com for flag: host
> 2021/03/31 16:20:16 Using default value: HssOriginRealm.com for flag: realm
> 2021/03/31 16:20:16 Using default value: HssDiameter for flag: product
> 2021/03/31 16:20:16 Using default value:  127.0.0.34:3868 for flag: addr
> 2021/03/31 16:20:16 Using default value: tcp for flag: network
> 2021/03/31 16:20:16 Using default value:  127.0.0.36:3869 for flag: laddr
> 2021/03/31 16:20:16 Using default value: magma-fedgw.magma.com for flag: dest_host
> 2021/03/31 16:20:16 Using default value: magma.com for flag: dest_realm
> 2021/03/31 16:20:16 Using default value: false for flag: disable_dest_host
> 2021/03/31 16:20:16 Using default value: false for flag: overwrite_dest_host
> 2021/03/31 16:20:16 failed to create S6aProxy: Invalid Diameter Address (tcp:// 127.0.0.34:3868): lookup  127.0.0.34: no such host

`/var/opt/magma/docker$ docker logs magmad`
> INFO:root:Starting magmad for UUID: a8580a44-3882-4c36-bbc1-818a22f323cc
> INFO:root:Starting collector...
> INFO:root:Streamer reconnect pause: 60
> INFO:root:Streamer timeout: 150
> INFO:root:Starting magmad...
> INFO:root:Listening on address 127.0.0.1:50052
> INFO:root:[SyncRPC] Opening stream to cloud
> INFO:root:Waiting before checking for updates for the first time...
> INFO:root:[SyncRPC] Waiting for requests
> WARNING:root:Collect s6a_proxy Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> INFO:root:Processing config update hssgw1
> control_proxy
> redis
> session_proxy
> s6a_proxy
> feg_hello
> csfb
> health
> swx_proxy
> eap_aka
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetOperationalStates Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> eap_sim
> aaa_server
> INFO:root:Checkin Successful! Successfully sent states to the cloud!
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> WARNING:root:Collect s6a_proxy Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> INFO:root:[SyncRPC] Got heartBeat from cloud
> INFO:root:Processing config update hssgw1
> ERROR:root:GetOperationalStates Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> INFO:root:Checkin Successful! Successfully sent states to the cloud!
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:Metrics upload error for service magmad (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service session_proxy (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service s6a_proxy (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service swx_proxy (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service eap_aka (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service eap_sim (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service aaa_server (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:Metrics upload error for service csfb (chunk 0)! [StatusCode.DEADLINE_EXCEEDED] Deadline Exceeded
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> ERROR:root:GetServiceInfo Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> WARNING:root:Collect s6a_proxy Error! [StatusCode.UNAVAILABLE] failed to connect to all addresses
> INFO:root:Checking for upgrade...
> INFO:root:There is work to be done:
>   current: latest
>   to_upgrade: 0.0.0-0
> Updating certificates in /etc/ssl/certs...
> INFO:root:[SyncRPC] Got heartBeat from cloud
> INFO:root:Processing config update hssgw1
> 1 added, 0 removed; done.
> Running hooks in /etc/ca-certificates/update.d...
> done.
> Cloning into 'magma'...
> ERROR:root:GetOperationalStates Error for s6a_proxy! [StatusCode.UNAVAILABLE] failed to connect to all addresses

**Comment by MGAbdElKhalek** [comment, marked as answer]
The connection is obtained after replacing 127.0.0.1 with "localhost"

**Reply by vik2595** [reply]
@MGAbdElKhalek 
Where did you replace 127.0.0.1 to localhost ?

**Reply by MGAbdElKhalek** [reply]
As far as I remember in the web page containing the s6a parameters in the federation GW web interface
